internal/rbd/rbd_util.go

```diff
@@ -979,11 +921,36 @@ func (rv *rbdVolume) getImageInfo() error {
 		return err
 	}
 	rv.imageFeatureSet = librbd.FeatureSet(features)
-	err = rv.updateVolWithImageInfo(rv.conn.Creds)
+
+	// Get parent information.
+	// TODO: Replace GetParentInfo() after
+	// https://github.com/ceph/go-ceph/issues/347 is fixed.
+	parentPool := make([]byte, 128)
+	parentName := make([]byte, 128)
+	parentSnapname := make([]byte, 128)
+	err = image.GetParentInfo(parentPool, parentName, parentSnapname)
+	if err != nil {
+		// Caller should decide whether not finding
+		// the parent is an error or not.
+		if strings.Contains(err.Error(), "No such file or directory") {
+			rv.ParentName = ""
+		} else {
+			return err
+		}
```

Comment:
why go-ceph is returning an error if the parent is not present. isn't it's expected to return success with an empty parent name?

Reply:
ceph is returning ENOENT: https://github.com/ceph/ceph/blob/103fd4af216820fefa12a2ec9576f9eb305fc05f/src/librbd/api/Image.cc#L244